File: python/tracing/dify/01_chat_blocking.py
"""Blocking Dify chat message traced by Respan."""

from _shared import DifyExampleRuntime, print_result


def main() -> None:
    workflow_name = "dify_chat_blocking.workflow"
    with DifyExampleRuntime(workflow_name) as runtime:
        client = runtime.chat_client()
        user = runtime.user("chat")
        response = client.create_chat_message(
            inputs={"city": "Paris"},
            query="Reply with one sentence about observability.",
            user=user,
            response_mode="blocking",
        )
        response.raise_for_status()
        result = response.json()
        summary = {"workflow": workflow_name, "answer": result.get("answer")}
        runtime.set_result(summary)
        print_result("chat_blocking", summary)


if __name__ == "__main__":
    main()

File: typescript/tracing/dify/01_chat_blocking.ts
#!/usr/bin/env node
import { ChatClient } from "dify-client";
import { withDifyRuntime } from "./_shared.js";

const workflowName = "dify_typescript_chat_blocking.workflow";

await withDifyRuntime(workflowName, async (runtime) => {
  const client = new ChatClient({
    apiKey: runtime.key("DIFY_CHAT_API_KEY"),
    baseUrl: runtime.baseUrl,
  });
  const response = await client.createChatMessage({
    inputs: { city: "Paris" },
    query: "Reply with one sentence about observability.",
    user: "respan-dify-ts-chat",
    response_mode: "blocking",
  });
  if (Symbol.asyncIterator in response) throw new Error("Expected blocking chat response");
  const data = response.data as Record<string, unknown>;
  const result = {
    status: response.status,
    answer: data.answer,
    conversationId: data.conversation_id,
  };
  runtime.setResult(result);
  return result;
});
